subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -399,18 +400,55 @@ static int cmd_proxy_solicit(const struct shell *sh, size_t argc,
 #endif /* CONFIG_BT_MESH_SHELL_GATT_PROXY */
 
 #if defined(CONFIG_BT_MESH_SHELL_PROV)
+static int ascii_decimal_to_le_bytes(const char *ascii_str, uint8_t *out, size_t *out_len)
+{
+	for (const char *p = ascii_str; *p; ++p) {
+		if (!isdigit((unsigned char)*p)) {
+			return -1; /* Invalid character */
+		}
+
+		uint16_t carry = *p - '0';
+
+		/* If this is the first digit and it's 0, we need at least one byte */
+		if (*out_len == 0) {
+			out[0] = carry;
+			*out_len = 1;
+			continue;
+		}
+
+		/* Multiply current result by 10 and add new digit */
+		for (size_t i = 0; i < *out_len; ++i) {
+			uint16_t value = out[i] * 10 + carry;
+
+			out[i] = value & 0xFF;
+			carry = value >> 8;
+		}
+
+		/* Handle overflow to next byte */
+		if (carry) {
+			if (*out_len >= PROV_IO_OOB_SIZE_MAX) {
+				return -1; /* Overflow */
+			}
+			out[(*out_len)++] = carry;
+		}
+	}
+
+	return 0;
+}
+
 static int cmd_input_num(const struct shell *sh, size_t argc, char *argv[])
 {
 	int err = 0;
-	uint32_t val;
+	uint8_t result[PROV_IO_OOB_SIZE_MAX] = {0};
+	size_t result_len = 0;
 
-	val = shell_strtoul(argv[1], 10, &err);
+	err = ascii_decimal_to_le_bytes(argv[1], result, &result_len);
```

Comment:
Does this code really work?

`result_len` here is 0. But AFAIS `ascii_decimal_to_le_bytes` expects `result_len` to be equal to size of `result`.